src/components/POLine/LineLinkedInstances/hooks/useLinkedInstances/useLinkedInstances.js

```diff
@@ -84,18 +85,22 @@ export const useLinkedInstances = line => {
   }
 
   const { isLoading, data } = useQuery(
-    ['ui-orders', 'linked-instances', line.id, linkedInstanceIds],
+    ['ui-orders', 'linked-instances', line.id, linkedInstanceIds, linkedTitles],
     async () => {
       const { data: relationTypesData } = await fetchInstanceRelationTypes();
 
       return batchRequest(
         async ({ params: searchParams }) => {
           const { instances = [] } = await ky.get('inventory/instances', { searchParams }).json();
 
-          return instances.map(instance => ({
+          const linkedInstances = instances.map(instance => ({
```

Comment:
build object instead to remove find

Reply:
additionally you can remove receivingTitle here, and add it when you iterate through linkedTitles in return